umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif9/spatial_extent.clj

```diff
@@ -0,0 +1,72 @@
+(ns cmr.umm-spec.umm-to-xml-mappings.dif9.spatial-extent
+  "These are functions that create a data structure that
+   will make it easier to perform the vertical spatial translation
+   for UMM -> DIF 9 records.")
+
+(defn create-elevation-key
+  "This function looks at the passed in type and checks to see if the tye contains the values
+   of Altitude with Min or Max or Depth with Min or Max. If any of these match then I can
+   match the type with the DIF 9 key. Otherwise pass back nil."
+  [elevation-type]
+  (if (.contains elevation-type "Altitude")
+    (if (.contains elevation-type "Min")
+      (keyword (str "Minimum_Altitude"))
+      (if (.contains elevation-type "Max")
+        (keyword (str "Maximum_Altitude"))))
+    (if (.contains elevation-type "Depth")
+      (if (.contains elevation-type "Min")
+        (keyword (str "Minimum_Depth"))
+        (if (.contains elevation-type "Max")
+          (keyword (str "Maximum_Depth")))))))
+
+(defn- create-vertical-domain-maps
+  "Returns a vector that contains a number of maps with vertical domain data in it.
+   An example return vector:
+   [{:Maximum_Depth \"-100\",
+     :Minimum_Depth \"-1\",
+     :Minimum_Altitude \"0\",
+     :Maximum_Altitude \"100\"}
+    {:Minimum_Altitude \"0\",
+     :Maximum_Altitude \"100\"}]
+    Passed in is a vector of UMM vertical domains key and value:
+    [#cmr.umm_spec.models.umm_common_models.VerticalSpatialDomainType{:Type Minimum Altitude, :Value 0}
+     #cmr.umm_spec.models.umm_common_models.VerticalSpatialDomainType{:Type Maximum Altitude, :Value 100}
+     ...]
+    and an empty vector to start. This function is recursive and builds the vector map."
+  [vert-vec result]
+  (if-let [vert (first vert-vec)]
+    (if (empty? result)
+      ;; If the vector is empty, create a new map and add the key value pairs.
+      ;; call the function again with the next vertial domain map.
+      ;; If the key is nil then just call the function with the next item.
+      (if-let [key (create-elevation-key (:Type vert))]
+        (->> (assoc {} key (:Value vert))
+             (assoc result 0)
+             (create-vertical-domain-maps (drop 1 vert-vec)))
+        (create-vertical-domain-maps (drop 1 vert-vec) result))
```

Comment:
For performance reasons, you should use `recur` instead of just calling the function directly. I _think_ it's because Clojure doesn't tail-optimize if you don't

https://clojuredocs.org/clojure.core/recur